packages/cysync-core/src/pages/OnBoarding/JoystickTraining/Dialogs/index.tsx

```diff
@@ -20,15 +21,23 @@ export const JoystickTrainingDialog: React.FC = () => {
 
   const [state, setState, isFinalState] = useStateWithFinality(
     TrainJoystickStatus.TRAIN_JOYSTICK_INIT,
-    TrainJoystickStatus.TRAIN_JOYSTICK_CENTER,
+    TrainJoystickStatus.TRAIN_JOYSTICK_CENTER + 2,
   );
 
   const trainJoystick: DeviceTask<void> = async connection => {
     const app = await ManagerApp.create(connection);
-    await app.trainJoystick(s => {
-      if (s < TrainJoystickStatus.TRAIN_JOYSTICK_CENTER) setState(s);
+    await app.trainJoystick(async s => {
+      if (s < TrainJoystickStatus.TRAIN_JOYSTICK_LEFT) setState(s);
+      if (s === TrainJoystickStatus.TRAIN_JOYSTICK_LEFT) {
+        setState(s);
+        await sleep(600);
+        setState(TrainJoystickStatus.TRAIN_JOYSTICK_LEFT + 1);
+      }
     });
-    setState(TrainJoystickStatus.TRAIN_JOYSTICK_CENTER);
+    setState(TrainJoystickStatus.TRAIN_JOYSTICK_CENTER + 1);
+    await sleep(600);
+    setState(TrainJoystickStatus.TRAIN_JOYSTICK_CENTER + 2);
+    navigateTo(routes.onboarding.cardTraining.path, 3000);
```

Comment:
Use the navigation path defined on the line `51`. Also remove the `useEffect`, if we are redirecting directly after successful run.